### PR TITLE
Add Form

### PR DESCRIPTION
**Pull request by SaboorSohaib**
## Hi Dear :wave: 

**I implemented the following requirements:**

:gem: Create a method called current_user in your ApplicationController to make current user data available to all controllers.
   - It will return the first user from the database.
   - Don't worry about proper authentication at this time. You will add it in the upcoming projects.
 
:gem: Create forms to perform the following functions:
   - Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
   - Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).

:gem: Allow Users to add likes to Posts.

**Thank You.**

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -6,5 +6,45 @@ def index
...
+    @comment.save
+    redirect_to user_post_path
+  end
+
+  def create_like
+    @like = Like.new(author: current_user, post: Post.find(params[:id]))
+    @like.save
+    redirect_to user_post_path
+  end
+
```
Comment (DammyShittu) [inline app/controllers/posts_controller.rb]: - [ ] The code for creating comments and likes should be in the respective comments and likes controllers. Kindly use the `new` and `create` in the comments and likes controllers.

- [ ] Kindly remove these lines on 9 and 10:

  ```
  @comment = Comment.new
  @like = Like.new
  ```
  Move them to the `new` actions in their respective controllers.

File: config/routes.rb [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
   root "users#index"
+  get 'posts/new', to: 'posts#new', as: 'create_post'
+  post 'posts/create', to: 'posts#create'
+  get 'users/:user_id/posts/:id', to: 'posts#show', as: 'post'
+  post 'users/:user_id/posts/:id', to: 'posts#create_comment'
+  post 'users/:user_id/posts/:id/likes', to: 'posts#create_like', as: 'like'
+
   resources :users, only: [:index, :show] do
     resources :posts, only: [:index, :show]
   end
```
Comment (DammyShittu) [inline config/routes.rb]: - [ ] The like and comments routes should be wrapped within the posts resources and they should point to their controllers, here is an idea:

  ```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show, :create, :new] do
      resources :comments, only: [:create] do
      resources :likes, only: [:create]
    end
  end
  ```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @SaboorSohaib,

## Highlights
✔️ Posts can be created by the current_user

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (KDlamini) [reply app/controllers/posts_controller.rb]: Well resolved ✔️

File: config/routes.rb (hunk 2 above)
Comment (KDlamini) [reply config/routes.rb]: Well resolved ✔️

**Review by KDlamini — APPROVED** [review]
### Hi @SaboorSohaib 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️ The application controller contains `#current_user` method
✔️ Routes are set up correctly
✔️ Posts, comments, and likes actions are present

# Status: Approved ✔️ 

Great-job-smiley

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**